keystone_performance/load_test.py

```diff
@@ -304,6 +313,75 @@ def notify_done(self):
         self._done = True
 
 
+
+user_info_array = None
+
+def init_user_info(args):
+    global user_info_array
+    auth_url = '%s/v3/auth/tokens' % args.url
+
+    if args.user_domain_id:
+        user_domain_info = {'id': args.user_domain_id}
+    else:
+        user_domain_info = {'name': args.user_domain_name}
+    if args.project_id:
+        project_info = {'id': args.project_id}
+    else:
+        project_info = {'name': args.project_name}
+    if args.project_domain_id:
+        project_info['domain'] = {'id': args.project_domain_id}
+    else:
+        project_info['domain'] = {'name': args.project_domain_name}
+    auth_req_body = {
+        'auth': {
+            'identity': {
+                'methods': ['password'],
+                'password': {
+                    'user': {
+                        'name': args.username,
+                        'domain': user_domain_info,
+                        'password': args.password
+                    }
+                }
+            },
+            'scope': {
+                'project': project_info
+            }
+        }
+    }
+    auth_req_body = json.dumps(auth_req_body)
+
+    headers = {'content-type': 'application/json', 'Accept-Charset': 'UTF-8'}
+    r = requests.post(auth_url, data=auth_req_body, headers=headers)
+    if (r.status_code == 201):
+        token = r.headers['x-subject-token']
+    else:
+        raise Exception('Error: Failed to get keystone token for getting user info. Status code: ' + str(r.status_code))
```

Comment:
The code should be restructured so that it does

```
if r.status_code != 201:
   raise Exception(...)
```

Otherwise it's confusing since the rest of the code after this is run only if r.status_code == 201.